### ISSUE TITLE
zip file for download of multiple files appears corrupted

### ISSUE THREAD

**Issue by benedetto** [owncloud/core#16656]
### Steps to reproduce
1. Select multiple files in the web interface and hit "download" to get a zip file containing all the files
2. Download it to your computer
3. Try to open it on a Linux system

### Expected behaviour
The file should open/extract into the selected files

### Actual behaviour
If I try to open it with a Linux GUI program for archive handling ([Ark](https://en.wikipedia.org/wiki/Ark_%28software%29) in my case) I only get an error message telling me that the file cannot be opened because it is corrupted. If I instead try to extract it with the extraction function in the context menu of my file manager, the whole archive is extracted into a file that has the name and the content of the first file of the zip archive but the size of the whole archive.

If I use ````unzip download.zip```` on the command line instead, the archive is extracted correctly. This happens to me on both owncloud instances I administer and I had several Linux users complaining about the issue.

#### Server 1
**Operating system**:
Ubuntu 12.04

**Web server:**
Apache 2.4.7 (Ubuntu)

**Database:**
Mysql 5.5.43

**PHP version:**
5.5.25

**ownCloud version:** (see ownCloud admin page)
8.0.3-4

**Updated from an older ownCloud or fresh install:**
updated

**List of activated apps:**
 - activity
 - appframework
 - bookmarks
 - calendar
 - contacts
 - documents
 - files
 - files_external
 - files_locking
 - files_mv
 - files_pdfviewer
 - files_sharing
 - files_texteditor
 - files_trashbin
 - files_versions
 - firstrunwizard
 - gallery
 - news
 - search_lucene

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => '5252a2ecf1e01',
  'passwordsalt' => 'xxx',
  'datadirectory' => '/data/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.0.3.4',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxx',
  'dbpassword' => 'xxx',
  'installed' => true,
  'forcessl' => true,
  'theme' => '',
  'maintenance' => false,
  'log_type' => 'owncloud',
  'logfile' => '/var/log/owncloud.log',
  'logtimezone' => 'Europe/Berlin',
  'log_rotate_size' => 10485760,
  'overwritewebroot' => '/owncloud',
  'trusted_domains' => 
  array (
    0 => 'xxx',
  ),
  'secret' => 'xxx',
  'overwrite.cli.url' => '/owncloud',
);
```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

#### Server 2
**Operating system**:
Debian 7

**Web server:**
Apache 2.2.22 (Ubuntu)

**Database:**
Mysql 5.5.43

**PHP version:**
5.4.39

**ownCloud version:** (see ownCloud admin page)
8.0.3-4

**Updated from an older ownCloud or fresh install:**
updated

**List of activated apps:**
 - activity
 - bookmarks
 - calendar
 - contacts
 - documents
 - files
 - files_external
 - files_locking
 - files_pdfviewer
 - files_sharing
 - files_texteditor
 - files_trashbin
 - files_versions
 - gallery
 - music
 - search_lucene

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'passwordsalt' => 'xxx',
  'datadirectory' => '/ext/ocdata',
  'dbtype' => 'mysql',
  'version' => '8.0.3.4',
  'dbname' => 'owncloud45',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxx',
  'dbpassword' => 'xxx',
  'installed' => true,
  'instanceid' => '50772500ab8c7',
  'forcessl' => true,
  'maintenance' => false,
  'theme' => '',
  'log_type' => 'syslog',
  'logfile' => '',
  'loglevel' => '1',
  'trusted_domains' => 
  array (
    0 => 'xxx',
  ),
  'overwritewebroot' => '/owncloud',
  'overwrite.cli.url' => '/owncloud',
  'secret' => 'xxx',
  'forceSSLforSubdomains' => true,
);
```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:**
Firefox 

**Operating system:**
Several Ubuntu flavours and versions

### Logs
No suspicious log entries

**Comment by LukasReschke** [comment]
What version of ark are you using? Can you provide any information which libraries it uses? (maybe there is something in the help window?)

**Comment by benedetto** [comment]
In my case it's Ark 2.19 on KDE 4.14.2. According to apt it depends on 

- libarchive12
- libc6 (>= 2.14)
- libkdecore5
- libkdeui5
- libkfile4,
- libkhtml5
- libkio5
- libkonq5abi1
- libkparts4
- libkpty4,
- libqt4-dbus (>= 4:4.5.3)
- libqtcore4 (>= 4:4.7.0~beta1)
- libqtgui4 (>= 4:4.5.3)
- libstdc++6 (>= 4.1.1)

and recommends 

- bzip2
- p7zip-full
- unzip
- zip

I would guess that the archive handling is basically provided by libarchive12 and the recommended command line tools for the different zip formats.

**Comment by DeepDiver1975** [comment]
Another funny ZipStreamer case? @McNetic you might be interested? THX a lot

**Comment by PVince81** [comment]
Possible duplicate of https://github.com/owncloud/core/issues/18905

**Comment by LukasReschke** [comment]
Upstream: https://bugs.kde.org/show_bug.cgi?id=353266 + https://github.com/McNetic/PHPZipStreamer/issues/25

tl;dr: Not much we can do. Use another application to open the ZIP file.
